geth/api/backend_test.go

```diff
@@ -0,0 +1,250 @@
+package api
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/status-im/status-go/geth/node"
+	"github.com/status-im/status-go/geth/params"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBackendStartNodeConcurrently(t *testing.T) {
+	backend := NewStatusBackend()
+	config := params.NodeConfig{}
+	count := 2
+	resultCh := make(chan error)
+
+	var wg sync.WaitGroup
+	wg.Add(count)
+
+	for i := 0; i < count; i++ {
+		go func() {
+			resultCh <- backend.StartNode(&config)
+			wg.Done()
+		}()
+	}
+
+	// close channel as otherwise for loop never finishes
+	go func() { wg.Wait(); close(resultCh) }()
+
+	var results []error
+	for err := range resultCh {
+		results = append(results, err)
+	}
+
+	require.Contains(t, results, nil)
+	require.Contains(t, results, node.ErrNodeRunning)
```

Comment:
I know it will never happen, but is it possible that the first node dies and the second one starts successfully? This way `results` will be `[nil, nil]` but the error is somewhere else?

Reply:
nice! I didn't get the second point, but if it panics it would be clear!